plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/CastDataTypeTest.java

```diff
@@ -0,0 +1,109 @@
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.testing.sql.SqlExecutor;
+import io.trino.testing.sql.TestTable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+public class CastDataTypeTest
+{
+    private final List<TestCaseInput> testCaseInputs = new ArrayList<>();
+
+    private OptionalInt inputSize = OptionalInt.empty();
+    private TestTable testTable;
+
+    private CastDataTypeTest() {}
+
+    public static CastDataTypeTest create()
+    {
+        return new CastDataTypeTest();
+    }
+
+    public CastDataTypeTest addColumn(String columnName, String columnType, List<Object> inputValues)
+    {
+        if (inputSize.isEmpty()) {
+            inputSize = OptionalInt.of(inputValues.size());
+        }
+        checkArgument(inputSize.getAsInt() == inputValues.size(), "Expected input size: %s, but found: %s", inputSize.getAsInt(), inputValues.size());
+        testCaseInputs.add(new TestCaseInput(columnName, columnType, inputValues));
+        return this;
+    }
+
+    public CastDataTypeTest execute(SqlExecutor sqlExecutor, String tableNamePrefix)
+    {
+        checkState(!testCaseInputs.isEmpty(), "No test case inputs");
+        List<String> columnsWithType = new ArrayList<>();
+        for (TestCaseInput input : testCaseInputs) {
+            columnsWithType.add("%s %s".formatted(input.columnName(), input.columnType()));
+        }
+        String tableDefinition = columnsWithType.stream()
+                .collect(joining(", ", "(", ")"));
+
+        if (inputSize.isPresent()) {
+            List<String> rowToInsert = new ArrayList<>();
+            for (int i = 0; i < inputSize.getAsInt(); i++) {
+                List<String> row = new ArrayList<>();
+                for (TestCaseInput input : testCaseInputs) {
+                    row.add(String.valueOf(input.inputValues().get(i)));
+                }
+                rowToInsert.add(String.join(", ", row));
+            }
+            testTable = new TestTable(sqlExecutor, tableNamePrefix, tableDefinition, rowToInsert);
+        }
+        else {
+            testTable = new TestTable(sqlExecutor, tableNamePrefix, tableDefinition);
+        }
+        return this;
+    }
+
+    public String tableName()
+    {
+        return testTable.getName();
+    }
+
+    public record TestCaseInput(String columnName, String columnType, List<Object> inputValues)
+    {
+        public TestCaseInput
+        {
+            requireNonNull(columnName, "columnName is null");
+            requireNonNull(columnType, "columnType is null");
+            requireNonNull(inputValues, "inputValues is null");
```

Comment:
ImmutableList#copyOf

Reply:
elements could be null. So Ill use `List#copyOf` here